src/app/test-helpers.ts

```diff
@@ -1,16 +1,61 @@
+import { Component, Directive, Input, HostListener } from "@angular/core";
 import { of } from "rxjs";
 import { REPOSITORIES, ORG_STATS } from "./mock-data";
 import { RepoService } from "./repo.service";
 import { StatService } from "./stat.service";
+import { HomeComponent } from "./home/home.component";
+import { RepoDetailComponent } from "./repo-detail/repo-detail.component";
+import { SearchComponent } from "./search/search.component";
+import { Routes } from '@angular/router';
 
-export const repoServiceStub: Partial<RepoService> = {
+export class RepoServiceStub implements Partial<RepoService> {
   getRepos() {
     return of(REPOSITORIES);
   }
+
+  getRepo() {
+    return of(REPOSITORIES[0]);
+  }
+
+  searchRepos(term: string) {
+    switch(term) {
+      case "zagaku":
+        return of([REPOSITORIES[0]]);
+        break;
+      case "blog":
+        return of([REPOSITORIES[5], REPOSITORIES[6]]);
+        break;
+      default:
+        return of([]);
+    }
+  }
 };
 
 export const orgStatServiceStub: Partial<StatService> = {
   getOrgStats() {
     return of(ORG_STATS);
   }
 };
+
+export const testRoutes: Routes = [
+  { path: '', redirectTo: '/dashboard', pathMatch: 'full' },
+  { path: 'dashboard', component: HomeComponent },
+  { path: 'detail/:name', component: RepoDetailComponent },
+  { path: 'search', component: SearchComponent }
+];
+
+/* eslint-disable */
```

Comment:
Just out of curiosity, what's the reasoning behind disabling `eslint` here?

Reply:
If I remove it, `eslint` complains that the selector doesn't start with the `app-` prefix that normally precedes component selector names. Idk why it's an issue here since this isn't a component, but I got instructions for this test from the Angular website so I'm guessing it's right!